pkg/controller/bastion/actuator.go

```diff
@@ -110,3 +113,22 @@ func getDisk(ctx context.Context, gcpclient gcpclient.Interface, opt *Options) (
 	}
 	return disk, nil
 }
+
+func createGCPClientAndOptions(ctx context.Context, a *actuator, bastion *v1alpha1.Bastion, cluster *controller.Cluster) (gcpclient.Interface, *Options, error) {
+	serviceAccount, err := gcp.GetServiceAccount(ctx, a.Client(), v1.SecretReference{Namespace: bastion.Namespace, Name: constants.SecretNameCloudProvider})
+	if err != nil {
+		return nil, nil, fmt.Errorf("%w, failed to get serviceaccount", err)
+	}
+
+	gcpClient, err := gcpclient.NewFromServiceAccount(ctx, serviceAccount.Raw)
+	if err != nil {
+		return nil, nil, fmt.Errorf("%w, failed to create GCP client", err)
+	}
+
+	opt, err := DetermineOptions(bastion, cluster, serviceAccount.ProjectID)
+	if err != nil {
+		return nil, nil, fmt.Errorf("%w, failed to setup GCP options", err)
```

Comment:
copy paste error - `failed to setup GCP options`

Reply:
fixed